cpp/arcticdb/util/decimal.hpp

```diff
@@ -39,6 +45,14 @@ namespace arcticdb {
             [[nodiscard]] std::string to_string(int scale) const;
             [[nodiscard]] bool is_negative() const;
             [[nodiscard]] Decimal operator-() const;
+            /**
+             * Compare two decimals. Since the format does not keep decimal point position, the comparison is byte-wise.
+             * @example
+             *  Decimal("1") == Decimal("1.0") -> false
+             *  Decimal("100") == Decimal("1.00") -> true
```

Comment:
At the moment this decimal class is essentially just a glorified bitfield that requires an external scale, so I'm not sure that bitwise comparison is totally surprising. Obviously there are some questionable aspects to that particular strategy, but we've decided that compatibility is more important than design perfection.

Since I already wrote a floating-point decimal class we could potentially use that in cases where compatibility is not required as it will be easier to make it do sensible things numerically speaking.